wip: Migrate from greandao to sqldelight

Vinetos opened the pull request:
#6 

UPDATE 17/04: As the old code of the application is *very* messy. I have chosen to rewrite all denylist activities in this MR.
I know it may take a long time, but not doing it will postpone the problem.

Vinetos committed:
wip: Migrate from greandao to sqldelight

Vinetos committed:
wip: Start the new activity implementation

Vinetos committed:
wip: Replace blacklist activity with an basic new

Vinetos committed:
wip: Add edit activity for denylist item (can not modify / save)

Vinetos committed:
wip: Add edit activity for denylist item (can not modify / save)

papjul commented:
I would suggest using (on top of what you already did) SQLite from Requery: https://github.com/requery/sqlite-android

See the "Why?" section of the README.

Can be summed up as:

> Using this library, you can keep up to date with the latest versions of SQLite and provide a consistent version across OS versions and devices.

Vinetos committed:
wip: todo add click listener

papjul commented:
@Vinetos Do you need help with the PR?

papjul committed:
Fix #22 - Foreground service type crash on Android 14+ (#23)

Vinetos commented:
> @Vinetos Do you need help with the PR?

Actually, A lot of thing happened and make me forget this work-in-pogress.

As I remember, most of the functionality were working but not fully tested.
If you have time, that would be awesome !

papjul commented:
> As I remember, most of the functionality were working but not fully tested.

Do you use this branch as your daily app, or do you still use YACB? If you are using it since April, I believe it should be rather stable for the main purposes.

From what I read, you began rewriting all denylist activities, are all the features complete?

If my testing looks good, can I move on to some cleanup? The project is rather old and its base code needs some modernizing to ensure its reliability in the future. A rewrite (one file at a time) in Kotlin would also be nice for DX and to make the code safer from `NullPointerException`s.

Vinetos commented:
> > As I remember, most of the functionality were working but not fully tested.
> 
> Do you use this branch as your daily app, or do you still use YACB? If you are using it since April, I believe it should be rather stable for the main purposes.
> 
> From what I read, you began rewriting all denylist activities, are all the features complete?
> 
> If my testing looks good, can I move on to some cleanup? The project is rather old and its base code needs some modernizing to ensure its reliability in the future. A rewrite (one file at a time) in Kotlin would also be nice for DX and to make the code safer from `NullPointerException`s.

I use this version daily, yes.

To be honest, I can not remember all denylist activities features activities are completed or not.
I completely uninstall the app and start new tests now.

In any case, I am ready to start clean-up and start a rewrite to Kotlin

papjul commented on app/src/main/java/fr/vinetos/tranquille/InfoDialogHelper.java:
```diff
@@ -104,8 +104,8 @@ public static void showDialog(Context context, NumberInfo numberInfo,
-            context.startActivity(EditBlacklistItemActivity
-                    .getIntent(context, name, numberInfo.number));
+//            context.startActivity(EditBlacklistItemActivity
+//                    .getIntent(context, name, numberInfo.number));
```
I think you forgot something here

papjul committed:
Merge upstream changes (#24)

For some reason, Tranquille was forked from an old commit from 2021.
This PR adds all commits (mostly translations) between 2021 and 2024
from upstream.

Once merged, I will take a look at testing/modernizing the code base to
get a release ready with the new database backend ;)

---------

Co-authored-by: APL <apolo_br@protonmail.com>
Co-authored-by: haygcao <ngokelcho@outlook.com>
Co-authored-by: ssantos <ssantos@web.de>
Co-authored-by: Maite Guix <maite.guix@gmail.com>
Co-authored-by: GobinathAL <gobinathal8@gmail.com>
Co-authored-by: z Z z <ruslan.zeezu@gmail.com>
Co-authored-by: Jane Kong <myonjane@gmail.com>
Co-authored-by: Alexandre Díaz <dev@redneboa.es>
Co-authored-by: DJEBBARI.ABDELHAMID <ahd.radio@gmail.com>
Co-authored-by: LP <dankeschumy@hotmail.it>
Co-authored-by: DankXylese <dankxylese@slapmywang.com>
Co-authored-by: AHOHNMYC <lqwh2h2cwa@protonmail.com>
Co-authored-by: Artem <artem@molotov.work>
Co-authored-by: gallegonovato <fran-carro@hotmail.es>
Co-authored-by: Translator-3000 <weblate.m1d0h@8shield.net>
Co-authored-by: Xosé M <correoxm@disroot.org>
Co-authored-by: SC <lalocas@protonmail.com>
Co-authored-by: Micke Nilsson <mikni@proton.me>
Co-authored-by: Stefano Privato <9ae976b2-7d28-4d98-bdc2-39642f86fd34@crazyer.anonaddy.com>
Co-authored-by: Dan <denqwerta@gmail.com>
Co-authored-by: Ettore Atalan <atalanttore@googlemail.com>
Co-authored-by: Iginio Massari <94639044+darthmassari@users.noreply.github.com>
Co-authored-by: ZehRique <zehrique.dev@gmail.com>
Co-authored-by: Skrripy <rozihrash.ya6w7@simplelogin.com>
Co-authored-by: Kristoffer Grundström <swedishsailfishosuser@tutanota.com>
Co-authored-by: Milo Ivir <mail@milotype.de>
Co-authored-by: Gabriel Camargo <g.camargo@proton.me>
Co-authored-by: Tzvika <mmm_45@walla.com>
Co-authored-by: Yaron Shahrabani <sh.yaron@gmail.com>
Co-authored-by: Сергій <sergiy.goncharuk.1@gmail.com>
Co-authored-by: hugoalh <hugoalh@users.noreply.hosted.weblate.org>
Co-authored-by: Xynngh <xynngh@protonmail.com>

papjul committed:
Complete migration from GreenDAO to SQLDelight (#25)

- [x] Update dependencies and targetSDK to 35
- [x] Revert the refactoring of the denylist (only ~10% was done and
everything else was deleted and not working). I suggest we keep the old
(but working!) files, so we can get a release without GreenDAO asap, and
refactor denylist later
- [x] Rename "blacklist" to "denylist" everywhere, as you started doing
(+ in French)
- [x] Use [ReQuery SQLite](https://github.com/requery/sqlite-android)
(on top of SQLDelight) to provide a consistent SQLite version across OS
versions and devices
- [x] Fix a few bugs and crashes

Tested:
- [x] Denylist import
- [x] Denylist export
- [x] Denylist add
- [x] Denylist edit
- [x] Denylist delete
- [x] Denylist enable/disable
- [x] Call blocked when matching a denylist item rule (and denylist
enabled)
- [x] Pattern matching in the call log (correctly match and display the
denylist item name)

papjul commented:
I've been using a debug build for the last few weeks, and I think it's ready to be merged for a release.
What do you think?

Vinetos commented:
Yep !
Me too !
